google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java

```diff
@@ -1268,7 +1268,7 @@ private TableResult queryRpc(
 
     long numRows;
     Schema schema;
-    if (results.getSchema() == null && results.getJobComplete()) {
+    if ((results.getSchema() == null && results.getJobComplete()) || !results.getJobComplete()) {
```

Comment:
This block reads quite confusing now.

consider flipping it..

if (results.getJobComplete() && results.getSchema() != null) {
  schema = Schema.fromPb(results.getSchema());
  // Your logic below for numRows, numDml...
} else {
  JobId jobId = ....
....
}

Reply:
done